src/wp-includes/style-engine/class-wp-style-engine.php

```diff
@@ -0,0 +1,560 @@
+			return $parsed_styles;
+		}
+
+		// Collect CSS and classnames.
+		foreach ( static::BLOCK_STYLE_DEFINITIONS_METADATA as $definition_group_key => $definition_group_style ) {
+			if ( empty( $block_styles[ $definition_group_key ] ) ) {
+				continue;
+			}
+			foreach ( $definition_group_style as $style_definition ) {
+				$style_value = _wp_array_get( $block_styles, $style_definition['path'], null );
+
+				if ( ! static::is_valid_style_value( $style_value ) ) {
+					continue;
+				}
+
+				$parsed_styles['classnames']   = array_merge( $parsed_styles['classnames'], static::get_classnames( $style_value, $style_definition ) );
+				$parsed_styles['declarations'] = array_merge( $parsed_styles['declarations'], static::get_css_declarations( $style_value, $style_definition, $options ) );
+			}
+		}
+
+		return $parsed_styles;
+	}
+
+	/**
+	 * Returns classnames, and generates classname(s) from a CSS preset property pattern, e.g., '`var:preset|<PRESET_TYPE>|<PRESET_SLUG>`'.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $style_value      A single raw style value or css preset property from the $block_styles array.
+	 * @param array $style_definition A single style definition from BLOCK_STYLE_DEFINITIONS_METADATA.
+	 *
+	 * @return array[] An array of CSS classnames.
+	 */
+	protected static function get_classnames( $style_value, $style_definition ) {
+		if ( empty( $style_value ) ) {
+			return array();
+		}
+
+		$classnames = array();
+		if ( ! empty( $style_definition['classnames'] ) ) {
+			foreach ( $style_definition['classnames'] as $classname => $property_key ) {
+				if ( true === $property_key ) {
+					$classnames[] = $classname;
+				}
+
+				$slug = static::get_slug_from_preset_value( $style_value, $property_key );
+
+				if ( $slug ) {
+					/*
+					 * Right now we expect a classname pattern to be stored in BLOCK_STYLE_DEFINITIONS_METADATA.
+					 * One day, if there are no stored schemata, we could allow custom patterns or
+					 * generate classnames based on other properties
+					 * such as a path or a value or a prefix passed in options.
+					 */
+					$classnames[] = strtr( $classname, array( '$slug' => $slug ) );
+				}
+			}
+		}
+
+		return $classnames;
+	}
+
+	/**
+	 * Returns an array of CSS declarations based on valid block style values.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $style_value          A single raw style value from $block_styles array.
+	 * @param array $style_definition     A single style definition from BLOCK_STYLE_DEFINITIONS_METADATA.
+	 * @param array $options              {
+	 *     Optional. An array of options. Default empty array.
+	 *
+	 *     @type bool $convert_vars_to_classnames Whether to skip converting incoming CSS var patterns, e.g., `var:preset|<PRESET_TYPE>|<PRESET_SLUG>`, to var( --wp--preset--* ) values. Default `false`.
+	 * }
+	 *
+	 * @return array<string, string> An array of CSS definitions, e.g., array( "$property" => "$value" ).
```

Comment:
```suggestion
	 * @return string[] An associative array of CSS definitions, e.g., array( "$property" => "$value" ).
```
Applies to all instances in the PR (watch out for alignment issues when this change is made for `@param`).

Regex Search for `string\[\](.*?)associative` in `wordpress-develop/src` to see precedent.